pkg/actuators/machine/reconciler.go

```diff
@@ -39,6 +39,20 @@ func newReconciler(scope *machineScope) *Reconciler {
 func (r *Reconciler) create() error {
 	klog.Infof("%s: creating machine", r.machine.Name)
 
+	if instances, err := r.getMachineInstances(); err == nil && len(instances) > 0 {
+		klog.Infof("%s: found existing instance %s for machine", r.machine.Name, aws.StringValue(instances[0].InstanceId))
+		// If we got here, then Exists failed to find the instance, and we were asked to create a new instance.
+		// The instance already exists, so requeue and start the reconcile again, Exists should pass now.
+		// Don't bother updating the status, Update will configure everything on the next reconcile.
+		return fmt.Errorf("%s: Possible eventual-consistency discrepancy; returning an error to requeue", r.machine.Name)
+	}
+	if r.providerStatus.InstanceID != nil && *r.providerStatus.InstanceID != "" {
```

Comment:
I believe we always patch the spec first, so on tight reconcile loops, it's possible for this to be nil.

Reply:
My test case showed what you are talking about, but has been mitigated by [commit 4](https://github.com/openshift/machine-api-provider-aws/pull/11/commits/e4510232e3cc860c9bb06edfe7527e1dd3a9e24f).

The logic has been updated (and tests accordingly) to show prevent us from changing the spec during `create`, we only update the status and with that fourth commit, we detect that we don't need a spec update, and don't send the spec update. Meaning the only update we send is for the status, which will set the instance ID.